bin/echo.c

```diff
@@ -135,13 +182,15 @@ int negotiate(struct s2n_connection *conn, int fd)
     }
 
     printf("Cipher negotiated: %s\n", s2n_connection_get_cipher(conn));
-    if (s2n_connection_is_session_resumed(conn)) {
+
+    bool session_resumed = s2n_connection_is_session_resumed(conn);
+    if (session_resumed) {
         printf("Resumed session\n");
     }
```

Comment:
Along with this, we could print the TLS session ticket, openssl prints the following in the debug mode: 

```
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_CHACHA20_POLY1305_SHA256
    Session-ID: D259EABA19140C4CF58D209EC2DE570B1FE4CD1A9FF972C37C3F25A8120EEFC8
    Session-ID-ctx: 
    Resumption PSK: 22CFF08AE14ACBA16CFB526E93901FE1021654E36D7A6DF1B3FDFF1FE04FA69C
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 46800 (seconds)
    TLS session ticket:
    0000 - 32 30 31 36 2e 30 37 2e-32 36 2e 31 35 00 00 00   2016.07.26.15...
    0010 - d5 72 16 17 d5 2b ec e9-9e 4b 3d 3c 53 d3 7b 5e   .r...+...K=<S.{^
    0020 - 15 66 90 e1 b3 4a f9 4c-47 32 77 99 48 9f cf 79   .f...J.LG2w.H..y
    0030 - 46 ef 42 f2 b7 fd 18 83-66 4a 23 4a 77 65 1f 40   F.B.....fJ#Jwe.@
    0040 - 5f b7 a4 5e ff ab 87 46-9d cd e1 f7 72 4e e8 66   _..^...F....rN.f
    0050 - 6e 40 94 4c ef 67 bc c5-8d 09 25 bb 15 5b d3 ed   n@.L.g....%..[..
    0060 - ee 20 23 f2 ac 5f e9 b8-71                        . #.._..q
    Start Time: 1620340134
    Timeout   : 7200 (sec)
    Verify return code: 21 (unable to verify the first certificate)
    Extended master secret: no
    Max Early Data: 0
 ```

Reply:
We actually couldn't print that ATM. We want customers to treat tickets as mysterious opaque blobs, so there are no methods to interrogate them. Adding extra debug info could be a separate issue if we decide it's valuable.